spi_nand_flash/src/nand.c

```diff
@@ -310,6 +312,23 @@ esp_err_t spi_nand_flash_read_sector(spi_nand_flash_device_t *handle, uint8_t *b
     return ret;
 }
 
+#if CONFIG_NAND_FLASH_VERIFY_WRITE
+static esp_err_t s_verify_write(spi_nand_flash_device_t *handle, dhara_sector_t sector_id, const uint8_t *expected_buffer, dhara_error_t *err)
+{
+    esp_err_t ret = ESP_OK;
+    if (dhara_map_read(&handle->dhara_map, sector_id, handle->read_buffer, err)) {
+        ret = ESP_ERR_FLASH_BASE + *err;
+        ESP_LOGE(TAG, "failed to read nand flash to verify previous write, err: 0x%x", *err);
+        return ret;
+    }
+
+    if (memcmp(handle->read_buffer, expected_buffer, handle->page_size)) {
```

Comment:
To make these errors actually useful for troubleshooting, I think it would be better to log the expected and actual word which doesn't match.

https://github.com/espressif/esp-idf/blob/9106c43accd9f5e75379f62f12597677213f5023/components/spi_flash/esp_flash_api.c#L985-L992